application/src/main/java/com/fiap/burger/application/config/ApplicationConfiguration.java

```diff
@@ -34,7 +35,7 @@ public ProductUseCase productUseCase(ProductGateway productGateway) {
     }
 
     @Bean
-    public PaymentUseCase paymentUseCase(PaymentGateway paymentGateway, OrderGateway orderGateway, ProductGateway productGateway, ClientGateway clientGateway) {
-        return new DefaultPaymentUseCase(paymentGateway, new DefaultOrderUseCase(orderGateway, productGateway, clientGateway));
+    public PaymentUseCase paymentUseCase(PaymentGateway paymentGateway) {
+        return new DefaultPaymentUseCase(paymentGateway, new DefaultOrderGateway());
```

Comment:
Não há necessidade de instanciar um `new DefaultOrderGateway()` o Spring irá injetar a implementação da interface OrderGateway, ou seja, será criado um `DefaultOrderGateway  `se fizer no exemplo abaixo.
```suggestion
    public PaymentUseCase paymentUseCase(PaymentGateway paymentGateway, OrderGateway orderGateway) {
        return new DefaultPaymentUseCase(paymentGateway, orderGateway);
```